pennylane/pauli/pauli_arithmetic.py

```diff
@@ -1016,6 +1016,8 @@ def simplify(self, tol=1e-8):
         for pw, coeff in items:
             if abs(coeff) <= tol:
                 del self[pw]
+        if len(self) == 0:
+            self = PauliSentence({})
```

Comment:
What is a better way to make sure that when simplify() is removing everything from the PauliSentence, that we still retrieve an empty PauliSentence instead of just None?

Reply:
I thought that would already be the case. Why would it return `None`? It should have the same behaviour is a Python dict right?

```python
a = {1: 2, 2: 3, 3: 4, 4: 5}

for k, v in list(a.items()):
    del a[k]

print(a)
```
```pycon
{}
```

Reply:
From master:

```python
import pennylane as qml

a = qml.pauli.PauliSentence(
    {
        qml.pauli.PauliWord({0: "X", 1: "Y"}): 1.2,
        qml.pauli.PauliWord({0: "X", 1: "Z"}): 0.5,
        qml.pauli.PauliWord({0: "Y", 1: "Z"}): 2.0,
        qml.pauli.PauliWord({0: "Z", 1: "X"}): -1.0,
    }
)

for k, v in list(a.items()):
    del a[k]

print(a)
```
```pycon
0 * I
```

Reply:
Yes you are right, I could swear there were cases where I was getting "None" but perhaps that was just from accidentally looking at `Hamiltonian` instances. I will remove this